Menu Level

ChielBruin opened the pull request:
Added a MenuLevel that is automatically loaded from a file when the game starts.
In this level are a key and a door and some new entities:
- WallFrame: Displays an image on a panel.
- PlayerTrigger: Triggers an action when the player is closer than a certain distance (in this case starts the main game).
  - This action can be loaded from an external file using the ScriptLoader class, or created manually.

Most interesting commits:
[c6f8f15] Addition of WallFrame
[c959e9d] Addition of PlayerTrigger
[1362c52] Addition of the ScriptLoader
[77bb38e] Changes to the Roomloader to support the newly added entities

Closes #154, closes #157, closes #158, closes #155 and closes #156.

ChielBruin committed:
Added a wall frame entity

This entity can show an image.

ChielBruin committed:
Added Direction enum

ChielBruin committed:
Small change to keep the tests running

ChielBruin committed:
Added playerTrigger entity

This entity triggers an action when the player is nearby.

ChielBruin committed:
Added some testing lines

The lines are comments so they do not break any tests.

ChielBruin committed:
Added ScriptLoader class

This class can load .class files from the scripts folder.
This is now used for making the PlayerTrigger start the game.

ChielBruin committed:
Cleaned up the code

- Added ScriptLoaderException to wrap some exceptions.
- Added generics to ScriptLoader to make it more versatile.
- Added Javadocs

ChielBruin committed:
Merge branch 'dev-sprint4' into feature-tut-level

ChielBruin committed:
Changed roomloader to support new entities

Now works with WallFrame and PlayerTrigger.
Also fixed all the tests that i broke.

ChielBruin committed:
Removed temporary method

ChielBruin committed:
Removed syso's

ChielBruin committed:
Added test for ScriptloaderException

ChielBruin committed:
Added tests for the WallFrame class

ChielBruin committed:
Fixed a test

Test started failing by a change of implementation.

ChielBruin committed:
Changed gitignores to include script files

ChielBruin committed:
Added tests for PlayerTrigger

ChielBruin committed:
Added tests for ScriptLoader

ChielBruin committed:
Merge branch 'dev-sprint4' into feature-tut-level

ChielBruin committed:
Added javadocs and fixed the building

Geweldig committed:
Fix new entities

This makes it possible for the new entities to be rendered on the
webpage. This also fixes the exception that would be thrown if the
entity was unknown, which means a fix like this will have a lower
priority in the future.

Geweldig commented on src/test/java/nl/tudelft/contextproject/controller/WaitingControllerTest.java:
```diff
@@ -37,9 +39,10 @@ public static void tearDownAfterClass() {
 	
 	/**
 	 * Create a new instance of the controller for each test.
+	 * @throws FileNotFoundException This should not happen.
```
Why shouldn't this happen? If you can guarantee this will never happen you can use SneakyThrows

ChielBruin replied:
After your comment i realised the throws is not needed so i will remove it.

Geweldig commented:
I have left one comment, other than that it looks really good and can be merged in my opinion!
If you could also add a readme file somewhere explaning how the crf format works that would be grand!

ChielBruin committed:
Removed some exception throwing

ChielBruin committed:
Merge branch 'feature-tut-level' of https://github.com/Taeir/ContextProject-MIGI2 into feature-tut-level

ChielBruin committed:
Added example file with explanation of .crf files

This file includes explanation about the used file format to store rooms.
The file also is an example of a valid (except the file name) room to load.

ChielBruin commented:
@BCrlrd I made the suggested changes, including a simple explanation of the file format in the documentation folder.

Geweldig commented on documentation/CaerbannogRoomFormat.crf:
```diff
@@ -0,0 +1,28 @@
+# <xPosition> <yPosition> <zPosition> <EntityType> [optionalValues]
+#
+# Lastly all the lights are written in the same way as entities but with the addition of a color.
+# This collor is in the following format: <r/g/b/a>, which results in this sintax for lights:
```
syntax*

Geweldig commented:
Looks good! Did find one typo, which is my only "complaint"!

ChielBruin committed:
Update CaerbannogRoomFormat.crf

Diocruel commented on src/main/java/nl/tudelft/contextproject/controller/GameController.java:
```diff
@@ -44,21 +44,22 @@ public GameController(SimpleApplication app, Level level) {
 		List<Light> lights = new ArrayList<>();
-		String[] tmp = file.getName().split("_")[0].split("x");
-		MazeTile[][] tiles = new MazeTile[Integer.parseInt(tmp[0])][Integer.parseInt(tmp[1])];
 		try {
```
This will probably have  a merge conflict with Taico's loader

Diocruel committed:
Merge branch 'dev-sprint4' into feature-tut-level

Conflicts:
	src/main/java/nl/tudelft/contextproject/controller/WaitingController.java
	src/test/java/nl/tudelft/contextproject/controller/WaitingControllerTest.java
	src/test/java/nl/tudelft/contextproject/roomIO/RoomReaderTest.java

Diocruel commented:
The branch should now be mergable
